command/edit.go

```diff
@@ -0,0 +1,136 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	cli "gopkg.in/urfave/cli.v1"
+	log "github.com/Sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	vault "github.com/seatgeek/hashi-helper/command/vault"
+)
+
+type preferences map[string]preference
+type preference struct {
+	KeybaseTeam  string `yaml:"keybase-team-name"`
+	Editor string `yaml:"editor"`
+}
+
+// EditEncryptedFile ...
+func EditEncryptedFile(c *cli.Context) error {
+
+	// identify the target file
+	if !c.Args().Present() {
+		return fmt.Errorf("Please provide a path and file name.")
+	}
+	filePath := c.Args().First()
+	if filePath == "" {
+		return fmt.Errorf("Empty path and file name.")
+	}
+
+	// get the preferences, define the editor and keybase team
+	var preferences preferences
+	configFile, err := vault.GetProfileConfig()
+	if err != nil {
+		return err
+	}
+	if err := yaml.Unmarshal(configFile, &preferences); err != nil {
+		return err
+	}
+	preference, ok := preferences["encrypted-file-config"]
+	if !ok {
+		return fmt.Errorf("No encrypted-file-config block found in config file.\n")
+	}
+	// editor
+	editor := os.Getenv("EDITOR")
+	if editor == "" {
+		editor = preference.Editor
+	}
+	if editor == "" {
+		editor = "pico"
```

Comment:
i would error if there is no `EDITOR` rather than default to `pico` (this won't work on Windows for example)

Reply:
👍 Sounds good.